source/MaterialXCore/Document.cpp

```diff
@@ -109,7 +109,7 @@ bool convertMaterialsToNodes(DocumentPtr doc)
                 }
                 else if (valueElement->isA<BindParam>())
                 {
-                    portChild = shaderNode->addParameter(valueElement->getName(), valueElement->getType());
+                    portChild = shaderNode->addInput(valueElement->getName(), valueElement->getType());
```

Comment:
The two if-cases of BindInput and BindParam can be merged as they do the same thing now. If the upgrade path works correctly I suppose BindParam is never used in a document, so maybe that case can just be removed here?

Reply:
As noted above, cleaning up to remove Material Elements is a separate task so more code will be removed.